### PR TITLE
fix(rp2040): fix subs registers wrong overflow handling

### LINKED ISSUE #46: wrong subs register overflow flag
catched by gdbdiff:
Register	StartValue	Emulator	Silicone
r0  		0x853d1eee	0x053d1eee	0x053d1eee
r1  		0x00020000	0x00020000	0x00020000
r2  		0xfffe0000	0xfffe0000	0xfffe0000
r3  		0x80000000	0x80000000	0x80000000
r4  		0x853d1eee	0x853d1eee	0x853d1eee
r5  		0xea718d61	0xea718d61	0xea718d61
r6  		0x00000002	0x00000002	0x00000002
r7  		0x00000001	0x00000001	0x00000001
r8  		0xffffffff	0xffffffff	0xffffffff
r9  		0x0000000d	0x0000000d	0x0000000d
r10  		0x0a7a3dc7	0x0a7a3dc7	0x0a7a3dc7
r11  		0x20000b60	0x20000b60	0x20000b60
r12  		0x2000017d	0x2000017d	0x2000017d
sp  		0x20041f90	0x20041f90	0x20041f90
lr  		0x10001a81	0x10001a81	0x10001a81
pc  		0x200001cc	0x200001ce	0x200001ce
xPSR* 		0x81000000	0x31000000	0x21000000
Flags		 [N---] 	 [--CO] 	 [--C-]

0x200001cc <__udivsi3+80>:	c0 1a	subs	r0, r0, r3

### PR DESCRIPTION
**Pull request by Turro75**
close #46

### CODE REVIEW

File: src/rp2040.ts [hunk 1]
```diff
@@ -1485,10 +1485,10 @@ export class RP2040 {
       const operand2 = this.registers[Rm] + (this.C ? 0 : 1);
       const result = (operand1 - operand2) | 0;
       this.registers[Rdn] = result;
-      this.N = (operand1 | 0) < (operand2 | 0);
+      this.N = !!(result & 0x80000000);
```
Comment (urish) [inline src/rp2040.ts:1488]: Why not `result < 0`? I think it's a bit easier to read it that way (and we also calculate `N` like that in other places, such as the `CMP` instruction

**Review by urish — CHANGES_REQUESTED** [review]
Thanks Valerio! It's very helpful that you find all these sneaky bugs.

Any chance you can add test cases for the SBCS / CMP fixes as well?

File: src/instructions.spec.ts [hunk 2]
```diff
@@ -1239,27 +1300,28 @@ describe('Cortex-M0+ Instruction Set', () => {
...
+        expect(cpu.rp2040.pendingSVCall).toEqual(true);
+      }
+
+      await cpu.singleStep(); // SVCall handler should run here
+      const registers2 = await cpu.readRegisters();
+      if (cpu instanceof RP2040TestDriver) {
+        expect(cpu.rp2040.pendingSVCall).toEqual(false);
+      }
+      expect(registers2.pc).toEqual(SVCALL_HANDLER + 2);
+      expect(registers2.r0).toEqual(0x55);
```
Comment (urish) [inline src/instructions.spec.ts:1323]: can you please explain this change?

File: src/instructions.spec.ts [hunk 3]
```diff
@@ -138,7 +138,7 @@ describe('Cortex-M0+ Instruction Set', () => {
   it('should execute `adcs r5, r4` instruction and set negative/overflow flags', async () => {
     await cpu.setPC(0x20000000);
     await cpu.writeUint16(0x20000000, opcodeADCS(r5, r4));
-    cpu.setRegisters({
+    await cpu.setRegisters({
```
Comment (urish) [inline src/instructions.spec.ts:141]: good catch!

**Comment by urish** [comment]
Also, can you move the STR fix into a different pull request (different branch), together with the test case from #43? 

I think a bunch of other instructions have the same issue as well: STRH, STRB, LDR, LDRB, LDRH, LDRS, LDRSB

We can skip creating tests for each one of them, but since we already have the test case for `STR`, let's at least include it

**Comment by Turro75** [comment]
I'm gonna reworking it

**Comment by urish** [comment]
Thank you!

**Pull request closed without merging**